### ISSUE TITLE
'Ghost' pods are getting registered

### ISSUE THREAD

**Issue by legendary-cookie** [AzisabaNetwork/Kuvel#8]
I have a deployment for lobby servers. I noticed the plugin registers new servers in velocity every few seconds, that don't exist (replicacount is 2).

You can see this in these logs, is this normal? Also sometimes after removing a pod from the deployment and then adding it back, I cannot connect anymore.
```
[16:42:46 INFO] [net.azisaba.kuvel.Kuvel]: Unregistered server: lobby-3 (d023768e-7724-479b-8b7f-81bdb47d78f0)
[16:43:16 INFO]: [connected player] 1vento (/10.40.2.130:54412) has disconnected
[16:43:16 INFO]: [server connection] 1vento -> lobby-1 has disconnected
[16:43:18 INFO]: [connected player] 1vento (/10.40.2.130:55928) has connected
[16:43:18 INFO]: [server connection] 1vento -> lobby-2 has connected
[16:43:28 INFO] [net.azisaba.kuvel.Kuvel]: Registered server: lobby-3 (7d8c9c8c-2e44-4656-9634-f12adc63b521)
[16:43:28 INFO] [net.azisaba.kuvel.Kuvel]: Registered server: lobby-4 (1cd211e3-bac8-42d3-b27f-3a3b3a2864c7)
[16:43:28 INFO] [net.azisaba.kuvel.Kuvel]: Unregistered server: lobby-3 (7d8c9c8c-2e44-4656-9634-f12adc63b521)
[16:43:28 INFO] [net.azisaba.kuvel.Kuvel]: Registered server: lobby-3 (d023768e-7724-479b-8b7f-81bdb47d78f0)
[16:43:28 INFO] [net.azisaba.kuvel.Kuvel]: Registered server: lobby-5 (ce54dec6-6b4c-42bb-86a1-7c2be192716d)
[16:43:28 INFO] [net.azisaba.kuvel.Kuvel]: Unregistered server: lobby-5 (ce54dec6-6b4c-42bb-86a1-7c2be192716d)
[16:43:28 INFO] [net.azisaba.kuvel.Kuvel]: Unregistered server: lobby-4 (1cd211e3-bac8-42d3-b27f-3a3b3a2864c7)
[16:43:28 INFO] [net.azisaba.kuvel.Kuvel]: Unregistered server: lobby-3 (d023768e-7724-479b-8b7f-81bdb47d78f0)
[16:43:38 INFO] [net.azisaba.kuvel.Kuvel]: Failed to add load balancer. Server name already occupied: lobby
```

**Comment by legendary-cookie** [comment]
BTW: I am not using the redis loadbalancer here, but the errors are also happening when using it.

**Comment by siloneco** [comment]
Thank you for letting us know. But tbh, I have no idea why this occurs. [Kuvel uses Watch](https://github.com/AzisabaNetwork/Kuvel/blob/main/src/main/java/net/azisaba/kuvel/discovery/impl/SimpleServerDiscovery.java#L38-L56) to finding new pods so never happens unless Kubernetes is firing the Watch event by mistake, I think.
However, it seems to be actually happening, so I would like to verify this in my environment when I have time.

**Comment by siloneco** [comment]
Hi, I checked the issue in my environment and unfortunately the same bug did not occur. If you can tell us what OS you are using on the nodes, k8s distribution, manifests, etc., as much as you can disclose, we will investigate further.

By the way, the LoadBalancer feature can be used without Redis, since Redis is only used to synchronize the association between pods and server names across multiple Velocity proxies. If you want to disable this feature, simply delete the labels from your ReplicaSet ( normally from Deployment or StatefulSet )
```yml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: some-server
  labels:
    minecraftServiceDiscovery: "true" # This is required to enable the load balancer feature. Delete this if you want to disable it.
    minecraftServerName: "some-server" # This will be the name of the load balancer
spec:
  replicas: 3
  selector:
    matchLabels:
      app: some-server
  template:
    metadata:
      labels:
        app: some-server
        minecraftServiceDiscovery: "true" # This is required to register a Pod with Velocity
        minecraftServerName: "some-server" # This will be the prefer name for the server
    spec:
      containers:
        - name: some-server
          image: itzg/minecraft-server:java8
          ports:
            - containerPort: 25565
          ...
```

**Comment by legendary-cookie** [comment]
All my nodes use Debian 10 or 11. The cluster is on version 1.22.2, set up using Kubespray, 3 masters 5 workers.
I use this these manifests for deploying: https://github.com/legendary-cookie/minecraft-k8
Anything else?

**Comment by siloneco** [comment]
The use of "watch" with the Kubernetes API sometimes caused unknown behavior, but this problem should be resolved in version 2.0.0 because I have changed the method to periodically do a get.
Please re-open this issue if the problem still occurs in the latest version.
